homeassistant/components/tuya/entity.py

```diff
@@ -256,7 +267,14 @@ def get_dptype(
             order = ["function", "status_range"]
         for key in order:
             if dpcode in getattr(self.device, key):
-                return DPType(getattr(self.device, key)[dpcode].type)
+                current_type = getattr(self.device, key)[dpcode].type
+                try:
+                    return DPType(current_type)
+                except ValueError:
+                    """Sometimes, we get ill-formed DPTypes from the cloud,
+                    this fixes them and maps them to the correct DPType.
+                    """
```

Comment:
This should be a comment not a docstring
```suggestion
                    # Sometimes, we get ill-formed DPTypes from the cloud,
                    # this fixes them and maps them to the correct DPType.
```